File: 백준/Silver/11441. 합 구하기/합 구하기.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt"
const input = fs.readFileSync(filePath)
  .toString()
  .trim()
  .split("\n")


function solution([n, digits, ...args]) {
  n = +n
  digits = digits.split(' ').map(Number)
  let [m, ...arg] = args
  m = +m
  arg = arg.map(x => x.split(' ').map(Number))
  let dp = [digits[0]]
  for(let i = 1; i < n; i++){
    dp[i] = dp[i - 1] + digits[i]
  }
  for(let i = 0; i < m ; i++){
    let [left, right] = arg[i]
    console.log(dp[right - 1] - (dp[left - 2] ?? 0))
  }
}

solution(input)

File: 백준/Silver/11441. 합 구하기/합 구하기.kt
import java.io.BufferedReader
import java.io.File
import java.io.InputStreamReader
import java.util.*

fun main() {
    val os = System.getProperty("os.name").lowercase(Locale.getDefault())
    val args: List<String> = if (os.contains("linux")) {
        val br = BufferedReader(InputStreamReader(System.`in`))
        br.readLines()
    } else {
        val inputFile = File("src/main/kotlin/input.txt")
        inputFile.readLines()
    }

    val n = args[0].toInt()
    val digits = args[1].split(' ').map { it.toInt() }
    val m = args[2].toInt()
    val arg = args.slice(3 until args.size).map { it -> it.split(' ').map { it.toInt() } }
    val dp = mutableListOf(digits[0], digits[0] + digits[1])
    for (i in 2 until n) {
        dp.add(dp[i - 1] + digits[i])
    }
    val result = mutableListOf<Int>()
    for (i in 0 until m) {
        val (left, right) = arg[i]
        result.add(dp[right - 1] - dp.getOrElse(left - 2) { 0 })
    }
    println(result.joinToString("\n"))

}
